common/data_refinery_common/models/models.py

```diff
@@ -784,17 +784,30 @@ def calculate_size(self) -> None:
         self.size_in_bytes = os.path.getsize(self.absolute_file_path)
         return self.size_in_bytes
 
-    def delete_local_file(self):
+    def delete_local_file(self, force=False):
         """ Deletes a file from the path and actually removes it from the file system,
         resetting the is_downloaded flag to false. Can be refetched from source if needed. """
-        if not settings.RUNNING_IN_CLOUD:
+        if not settings.RUNNING_IN_CLOUD and not force:
             return
 
         try:
             os.remove(self.absolute_file_path)
         except OSError:
             pass
 
+    def delete_s3_file(self, force=False):
+        if not settings.RUNNING_IN_CLOUD and not force:
```

Comment:
Since you are deleting a file in S3, it is supposed to be:
```
if not settings.RUNNING_IN_CLOUD or not force:
```

Reply:
(I am not familiar with the calling context of this method, so my comment above could be totally wrong.)

Reply:
`force` defaults to false so `or not force` would make the conditional true most of the time. The idea here is that if RUNNING_IN_CLOUD is false but we want to force the S3 code anyway, then we shouldn't return early. (I agree the conditional  is a bit confusing because it's dealing with a case where we want to exit the function early rather than only running the function if it ends up being true.

Reply:
Thanks. It will be great if you can add some docstring to clarify its usage.